### PR TITLE
Fix metadata from TVDB

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: medusa/indexers/tvdbv2/tvdbv2_api.py [hunk 1]
```diff
@@ -329,9 +332,9 @@ def _parse_images(self, sid):
 
         for image_type, image_count in self._object_to_dict(series_images_count).iteritems():
             try:
-
                 if search_for_image_type and search_for_image_type != image_type:
-                    continue
+                    if image_type == 'poster' and search_for_image_type != 'poster_thumb':
+                        continue
```
Comment (medariox) [inline medusa/indexers/tvdbv2/tvdbv2_api.py]: We need to use posters for 'poster_thumb' (TVDB doesn't have any poster_thumbs).

File: medusa/indexers/tvdbv2/tvdbv2_api.py [hunk 2]
```diff
@@ -354,7 +357,7 @@ def _parse_images(self, sid):
                         if k is None or v is None:
                             continue
 
-                        k, v = k.lower(), v.lower()
+                        # k, v = k.lower(), v.lower()
                         _images[image_type][image.resolution][bid][k] = v
```
Comment (medariox) [inline medusa/indexers/tvdbv2/tvdbv2_api.py]: That one was sneaky. Some values are int and can't use lower().

File: medusa/indexers/indexer_base.py [hunk 3]
```diff
@@ -90,7 +90,7 @@ def __init__(self,
         self.config['session'] = session if session else requests.Session()
 
         self.config['banners_enabled'] = banners
-        self.config['image_type'] = None
+        self.config['image_type'] = image_type
         self.config['actors_enabled'] = actors
```
Comment (medariox) [inline medusa/indexers/indexer_base.py]: Caused image_type to be always None.

File: medusa/indexers/tvdbv2/tvdbv2_api.py [hunk 4]
```diff
@@ -233,7 +236,7 @@ def _get_episodes(self, tvdb_id, specials=False, aired_season=None):  # pylint:
         for cur_ep in episodes:
             if self.config['dvdorder']:
                 log().debug('Using DVD ordering.')
-                use_dvd = cur_ep['dvd_season'] != None and cur_ep['dvd_episodenumber'] != None
+                use_dvd = cur_ep['dvd_season'] is not None and cur_ep['dvd_episodenumber'] is not None
```
Comment (fernandog) [inline medusa/indexers/tvdbv2/tvdbv2_api.py]: @medariox didn't we agreed with != ?

File: medusa/indexers/tvdbv2/tvdbv2_api.py (hunk 2 above)
Comment (fernandog) [inline medusa/indexers/tvdbv2/tvdbv2_api.py]: wont break this? `k.endswith('path'):` and other possible string checks?

Comment (fernandog) [inline medusa/indexers/tvdbv2/tvdbv2_api.py]: can't we ?  `k, v = k.lower() if not isinstance(k, int) else k , ....`

File: medusa/indexers/tvdbv2/tvdbv2_api.py (hunk 4 above)
Comment (medariox) [inline medusa/indexers/tvdbv2/tvdbv2_api.py]: I hope not.

**Pull request merged**